netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -331,6 +337,12 @@ protected boolean isGracefulShutdownComplete() {
    */
   private void createStream(CreateStreamCommand command, final ChannelPromise promise)
           throws Exception {
+    if (goAwayStatus != null) {
```

Comment:
I think I moved it while trying to find the bug I ended up fixing with GracefulCloseCommand. At this point it just seems to logically make more sense. I'd be happy to revert it.